### DISCUSSION TITLE
java好像无法编译AbstractPermitteeId这个类

### DISCUSSION THREAD

**Discussion by pilgrim773** [Samarium150/mirai-console-lolicon#40] [Q&A] [answered]
我想参考本项目用java写一个基于指令的插件，但是编译器一直报
```
错误: 无法访问AbstractPermitteeId
import net.mamoe.mirai.console.permission.AbstractPermitteeId;
                                         ^
  错误的类文件: C:\Users\hisou\.gradle\caches\modules-2\files-2.1\net.mamoe\mirai-console\2.6.2\5b43b7a026b33c2c69ff361cf1d7de7937afe99b\mirai-console-2.6.2.jar(/net/mamoe/mirai/console/permission/AbstractPermitteeId.class)
    类文件包含格式错误的变量元数方法: AbstractPermitteeId(PermitteeId[],DefaultConstructorMarker)
    请删除该文件或确保该文件位于正确的类路径子目录中。
```

代码很简单就是去获取一下权限
`PermissionService.Companion.permit(AbstractPermitteeId.AnyContact.INSTANCE,
                PermissionId.parseFromString("*:*"));`

mirai那边官方说我版本问题，不过我已经试过2.6.2和2.6.3都不行，很难受
本来以为是我自己项目的gradle配置有问题，但是直接在本项目里建src/main/java再里面写，还是报错

**Comment by Samarium150** [comment, marked as answer]
确实能复现，不过我看到你在官方那关闭了issue？
直接从IDEA新建项目再加一行
`import net.mamoe.mirai.console.permission.AbstractPermitteeId;`
就能复现
建议重开issue反馈一下

**Reply by pilgrim773** [reply]
我看没人解决就给关了，再开一下好了
感觉是官方确实缺失了对java的适配

另外想问一下我已经有chat-command这个插件了，但是如果不加这句权限获取代码，就仍然无法聊天里输入指令(控制台可以用)。是不是关键就在这句报错的代码上？

**Reply by Samarium150** [reply]
不，你可以直接修改`/config/Console/PermissionService.yml`
比如
```yaml
grantedPermissionMap: 
  '*:*': 
    - console
    - u123456789
```
这样qq号为123456789的用户就能在聊天里输入指令了
